resolve/index.go

```diff
@@ -118,7 +118,7 @@ type ruleRecord struct {
 //
 // kindToResolver is a map from rule kinds (for example, "go_library") to
 // Resolvers that support those kinds.
-func NewRuleIndex(mrslv func(r *rule.Rule, pkgRel string) Resolver, exts ...interface{}) *RuleIndex {
+func NewRuleIndex(mrslv func(ruleKind, pkgRel string) Resolver, exts ...interface{}) *RuleIndex {
```

Comment:
This is technically public API, although I don't see how a language implementation would use it. Can we keep its signature unchanged somehow?

Reply:
I was assuming no tests failing meant it wasn't public API 😢

Reply:
I don't. Let's ship and find out.